src/biogeophys/SectorWaterMod.F90

```diff
@@ -0,0 +1,1238 @@
+          real(r8), intent(out) :: dom_consumption_volr_limited( bounds%begg:bounds%endg)
+          
+          real(r8), intent(out) :: liv_demand_volr_limited( bounds%begg:bounds%endg)
+          real(r8), intent(out) :: liv_consumption_volr_limited( bounds%begg:bounds%endg)
+          
+          real(r8), intent(out) :: elec_demand_volr_limited( bounds%begg:bounds%endg)
+          real(r8), intent(out) :: elec_consumption_volr_limited( bounds%begg:bounds%endg)
+          
+          real(r8), intent(out) :: mfc_demand_volr_limited( bounds%begg:bounds%endg)
+          real(r8), intent(out) :: mfc_consumption_volr_limited( bounds%begg:bounds%endg)
+          
+          real(r8), intent(out) :: min_demand_volr_limited( bounds%begg:bounds%endg)
+          real(r8), intent(out) :: min_consumption_volr_limited( bounds%begg:bounds%endg)
+ 
+          !
+          ! !LOCAL VARIABLES:
+          integer  :: g  ! gridcell index
+          real(r8) :: available_volr ! volr available for withdrawal [m3]
+          real(r8) :: max_demand_supported_by_volr ! [kg/m2] [i.e., mm]
+          
+          ! ratio of demand_volr_limited to demand for each grid cell
+          real(r8) :: dom_demand_limited_ratio_grc(bounds%begg:bounds%endg)
+          real(r8) :: liv_demand_limited_ratio_grc(bounds%begg:bounds%endg)
+          real(r8) :: elec_demand_limited_ratio_grc(bounds%begg:bounds%endg)
+          real(r8) :: mfc_demand_limited_ratio_grc(bounds%begg:bounds%endg)
+          real(r8) :: min_demand_limited_ratio_grc(bounds%begg:bounds%endg)
+ 
+ 
+          character(len=*), parameter :: subname = 'CalcSectorDemandVolrLimited'
+          !-----------------------------------------------------------------------
+          
+
+          do g = bounds%begg, bounds%endg
+               if (volr(g) > 0._r8) then
+                    available_volr = volr(g) * (1._r8 - this%params%sectorwater_river_volume_threshold)
+                    max_demand_supported_by_volr = (available_volr / grc%area(g) * m3_over_km2_to_mm)
+               else
+                    ! Ensure that negative volr is treated the same as 0 volr
+                    max_demand_supported_by_volr = 0._r8
+               end if
+               
+               ! I think the algorithm is potentially too conservative
+               ! I will need to check river discharge when there is high amount of unsatisfied demand
+               ! The reason why I am saying this is because we compare the expected demand for the entire day with the current volume available in the river...
+               ! It would make sense to use such an algorithm under the condition that volr does not change much during a day for a given gridcell
+               ! But if this is not the case we maybe underestimate the amount of water available for usage.
+               ! If this would be done once a day, then no problem, but we do it at each time step
+               ! This means that volr get 
```

Comment:
Conservative because it's looking at the river once and saying "this is how much water we have." But 3 hours later (timescale of coupling) that river will have flowed some more. So it's basically underestimating the available water by a factor of 8. 

Seemingly this is how it's done for irrigation, but it's not as much of an issue there because irrigation is only applied over a 4-hour window (as opposed to 24 for other sectors' demand).

Calculating availability limitation at the 3-hour timestep seems tricky. For now, to see how much difference this might make,Sabin will test multiplying available water in the timestep by 8. Conservation errors will be handled, as for irrigation, by taking extra needed water from ocean.

@swensosc, do you have any thoughts on the best way to handle this?

Reply:
Conservative because it's looking at the river once and saying "this is how much water we have." But 3 hours later (timescale of coupling) that river will have flowed some more. So it's basically underestimating the available water by a factor of 8. 

Seemingly this is how it's done for irrigation, but it's not as much of an issue there because irrigation is only applied over a 4-hour window (as opposed to 24 for other sectors' demand).

Calculating availability limitation at the 3-hour timestep seems tricky. For now, to see how much difference this might make,Sabin will test multiplying available water in the timestep by 8. Conservation errors will be handled, as for irrigation, by taking extra needed water from ocean.

@swensosc, do you have any thoughts on the best way to handle this?

Reply:
@swensosc The factor of 8 is because we're looking at how much water is available in one 3-hour timestep, but demand is actually spread out over 24 hours. So 24/3 = 8.

Reply:
Just chiming into make sure the final solution we use allows for the coupling timestep from the ROF model to vary. For a case this is the ROF_NCPL and NCPL_BASE_PERIOD variables. This isn't normally sent to CTSM, as ROF is expected to do it's own thing and usually CTSM doesn't need to know. If CTSM needs to know the coupling, we might need to pass it into the CTSM namelist.